### ISSUE TITLE
Docker: missing goroot

### ISSUE THREAD

**Issue by agarwalayush** [gomods/athens#1992]
I am using the docker to run athens using the instructions mentioned [here](https://docs.gomods.io/). However, on every request to fetch the package, I encounter 500. Logs show that `go` binary is trimmed and the system can not find GOROOT.

**Error Message**
```
ubuntu@ip-10-1-5-90:~/walkthrough$ GO111MODULE=on go run .
go: github.com/athens-artifacts/samplelib@v1.0.0: reading http://localhost:3000/github.com/athens-artifacts/samplelib/@v/v1.0.0.mod: 500 Internal Server Error
```

<pre>
```INFO[5:37AM]: Exporter not specified. Traces won't be exported
INFO[5:37AM]: Starting application	tcpPort=:3000
DEBUG[5:39AM]: saving github.com/athens-artifacts/samplelib@v1.0.0 to storage...	http-method=GET http-path=/github.com/athens-artifacts/samplelib/@v/v1.0.0.mod request-id=502f804e-f757-4ee1-8d3d-8653d45e39ff
ERROR[5:39AM]: exit status 2: go: cannot find GOROOT directory: 'go' binary is trimmed and GOROOT is not set
	http-method=GET http-path=/github.com/athens-artifacts/samplelib/@v/v1.0.0.mod kind=Internal Server Error module= operation=download.VersionModuleHandler ops=[download.VersionModuleHandler pool.GoMod protocol.GoMod protocol.processDownload stash.Pool stasher.Stash stasher.fetchModule goGetFetcher.Fetch module.downloadModule] request-id=502f804e-f757-4ee1-8d3d-8653d45e39ff version=
INFO[5:39AM]: incoming request	http-method=GET http-path=/github.com/athens-artifacts/samplelib/@v/v1.0.0.mod http-status=500 request-id=502f804e-f757-4ee1-8d3d-8653d45e39ff
```
</pre>

**To Reproduce**
1.  `sudo docker run -d -p '3000:3000' gomods/athens:latest`
2. `git clone https://github.com/athens-artifacts/walkthrough.git`
3.  `sudo apt  install golang-go`
4. `GO111MODULE=on go run .`

**Expected behavior**
Packages should have been downloaded

**Environment (please complete the following information):**
 - OS: Linux 64bit, ubuntu24
 - Go version : unable to identify
 - Proxy version : latest
 - Storage (fs/mongodb/s3 etc.) : default/in-memory

**Additional context**

**Comment by pmalmsten** [comment]
+1, I also observed this last Friday, running from WSL 2 on Windows 11. Same 500 response and error in the Athens standard output.

```
PS C:\Users\paulmalmsten\GoModules\walkthrough> docker image inspect gomods/athens:latest
[
    {
        "Id": "sha256:d3c84d276df198872a229f0a437cc415c8abd3863ea3ed744d0526d5922e3666",
        "RepoTags": [
            "gomods/athens:latest"
        ],
        "RepoDigests": [
            "gomods/athens@sha256:9fa45e7c12ef3fc225f90293c8079ccd347f6e8acd4d4c2c643fe186de7295cc"
        ],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2024-09-30T16:56:14.330870663Z",
        "DockerVersion": "",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "3000/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "GO111MODULE=on"
            ],
            "Cmd": [
                "athens-proxy",
                "-config_file=/config/config.toml"
            ],
            "ArgsEscaped": true,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": [
                "/sbin/tini",
                "--"
            ],
            "OnBuild": null,
            "Labels": {
                "org.opencontainers.image.created": "2024-09-30T16:54:40.762Z",
                "org.opencontainers.image.description": "A Go module datastore and proxy",
                "org.opencontainers.image.licenses": "MIT",
                "org.opencontainers.image.revision": "74b3cf50828d50d2b27da0eb674582c74000eeeb",
                "org.opencontainers.image.source": "https://github.com/gomods/athens",
                "org.opencontainers.image.title": "athens",
                "org.opencontainers.image.url": "https://github.com/gomods/athens",
                "org.opencontainers.image.version": "v0.15.2"
            }
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 170126361,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/ce2ee6a7ff5f17ca79f86fea8a013886a2d1e2780cf00d242b7cda8975e20649/diff:/var/lib/docker/overlay2/527738e9e97f1d7d0351560d5881d09c211be9017ba0be57119a91ab76a73001/diff:/var/lib/docker/overlay2/710dffd410313134f67d77c9db7473c9d0bf32ba274b1aff023daefc561ce0ad/diff:/var/lib/docker/overlay2/7c305a72e977ffd70cec1ef8867c70e0055af2ffbdd6e2711e5118d2400cd027/diff:/var/lib/docker/overlay2/ad5a9210658acb1f786065ad4b31b2c89262717b6bfb2772bd7ff41f01fcf0ba/diff:/var/lib/docker/overlay2/94523c8b0aa2b1613673e677b82a0e15ad81963ccdd60d9efbbedb8b2b8a870a/diff",
                "MergedDir": "/var/lib/docker/overlay2/b6312cfbe69cf5366ef258da788273817c92251cc580a596273f1a02c3a0af08/merged",
                "UpperDir": "/var/lib/docker/overlay2/b6312cfbe69cf5366ef258da788273817c92251cc580a596273f1a02c3a0af08/diff",
                "WorkDir": "/var/lib/docker/overlay2/b6312cfbe69cf5366ef258da788273817c92251cc580a596273f1a02c3a0af08/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:5181257c7956c691316e978aef13e2365e12e5f164337a2b7910d829d657d6c5",
                "sha256:59401b27deee218be0a9876e18090ed847bb90741e552c7de72b8b513f8c60f4",
                "sha256:2674ad00bde69923492057e9e2d12c484625a419bd01e5793b2067c3a62a15a7",
                "sha256:725e1d8447dd2b79ea3eba385b877a81ed615cc76a616360d00be2831e03d046",
                "sha256:3500285dd5b118c7d41ca193cfd3d3c27cc664758541266501bc3c50fafff93e",
                "sha256:9685db17a4ede6b69e74a54554a7e583329221f2302bea8cb42a82d70211e57f",
                "sha256:c5cdf4f946e925a8933a0b30945422d4d911731e99e0e6e532f3140836e9ac83"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]
```

**Comment by pmalmsten** [comment]
As a workaround, I reverted to v0.15.1: `docker run -p '3000:3000' gomods/athens:v0.15.1`

**Comment by agarwalayush** [comment]
Looks like they are aware of the issue and have pushed the change #1989 which hasn't made to the release yet.

**Comment by matt0x6F** [comment]
Fixed as of v0.15.3
